subprojects/plugin-use/src/integTest/groovy/org/gradle/plugin/repository/ResolvingFromSingleCustomPluginRepositorySpec.groovy

```diff
@@ -48,22 +50,29 @@ class ResolvingFromSingleCustomPluginRepositorySpec extends AbstractDependencyRe
     }
 
     private String useCustomRepository(String repoType, PathType pathType) {
-        def repoUrl = 'Nothing'
-        if (repoType == MAVEN) {
-            repoUrl = PathType.ABSOLUTE.equals(pathType) ? mavenRepo.uri : mavenRepo.getRootDir().name
-        } else if (repoType == IVY) {
-            repoUrl = PathType.ABSOLUTE.equals(pathType) ? ivyRepo.uri : ivyRepo.getRootDir().name
-        }
+        def repoUrl = buildRepoPath(repoType, pathType)
         settingsFile << """
-          pluginRepositories {
+          pluginManagement {
+            repositories {
               ${repoType} {
                   url "${repoUrl}"
               }
+            }
           }
         """
         return repoUrl
     }
 
+    private String buildRepoPath(String repoType, PathType pathType) {
+        def repoUrl = 'Nothing'
+        if (repoType == MAVEN) {
+            repoUrl = PathType.ABSOLUTE.equals(pathType) ? mavenRepo.uri : mavenRepo.getRootDir().name
```

Comment:
I think this would be tidier by simply passing around a `Repository` or one of its subtypes around and use its `uri` property and other query methods.